src/config.rs

```diff
@@ -0,0 +1,182 @@
+use std::env::VarError;
+use std::error::Error;
+use std::fmt;
+use std::str::FromStr;
+
+const ENVIRONMENT_VARIABLE_PREFIX: &str = "MONITORBOT_";
+
+#[derive(Clone, Debug)]
+pub struct Config {
+    // http server port to bind to
+    pub port: u16,
+    // github api tokens to collect rate limit statistics
+    pub gh_rate_limit_tokens: String,
+    // github rate limit stats data cache refresh rate frequency (in seconds)
+    pub gh_rate_limit_stats_cache_refresh: u64,
+}
+
+impl Config {
+    pub fn from_env() -> Result<Self, ConfigError> {
+        Ok(Self {
+            port: Self::default_env("PORT", 3001)?,
+            gh_rate_limit_tokens: Self::require_env("RATE_LIMIT_TOKENS")?,
+            gh_rate_limit_stats_cache_refresh: Self::default_env::<u64>(
+                "GH_RATE_LIMIT_STATS_REFRESH",
+                120u64,
+            )?,
```

Comment:
Type inference should be able to figure the `u64`s out on its own, both for the `::<u64>` and `120u64`.

Reply:
sure.